processing/src/main/java/org/apache/druid/query/QueryRunnerFactoryConglomerate.java

```diff
@@ -19,9 +19,22 @@
 
 package org.apache.druid.query;
 
+import org.apache.druid.error.DruidException;
+
 /**
  */
-public interface QueryRunnerFactoryConglomerate
+public interface QueryRunnerFactoryConglomerate extends QueryToolChestWarehouse
 {
   <T, QueryType extends Query<T>> QueryRunnerFactory<T, QueryType> findFactory(QueryType query);
+
+  @Override
+  default <T, QueryType extends Query<T>> QueryToolChest<T, QueryType> getToolChest(QueryType query)
+  {
+    QueryRunnerFactory<T, QueryType> factory = findFactory(query);
+    if (factory == null) {
+      throw DruidException
+          .defensive("QueryRunnerFactory for QueryType [%s] is not registered!", query.getClass().getName());
+    }
+    return factory.getToolchest();
+  }
```

Comment:
Originally, the Warehouse and the Conglomerate were separated because the conglomerate was only needed on "leaf nodes".  When we introduced the idea of sub-queries, it made it true that brokers also need the conglomerate, which takes away the reason for having the objects separated.  I don't think this needs to have a default implementation, but instead we should just make the conglomerate be the only thing depended on.  The Warehouse should be deprecated, but continue to exist for a bit (one release) just to enable any extensions that depend on it to be updated to depend on the conglomerate instead.